core/Updates/2.14.0.php

```diff
@@ -0,0 +1,46 @@
+<?php
+/**
+ * Piwik - free/libre analytics platform
+ *
+ * @link http://piwik.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ *
+ */
+
+namespace Piwik\Updates;
+
+use Piwik\Config;
+use Piwik\Updater;
+use Piwik\Updates;
+use Piwik\Plugin\Manager;
+
+class Updates_2_14_0 extends Updates
+{
+    public function doUpdate(Updater $updater)
+    {
+        $this->uninstallPlugin('UserSettings');
+    }
+
+    private function uninstallPlugin($plugin)
+    {
+        $pluginManager = Manager::getInstance();
+
+        if ($pluginManager->isPluginInstalled($plugin)) {
+
+            if ($pluginManager->isPluginActivated($plugin)) {
+                $pluginManager->deactivatePlugin($plugin);
+            }
+
+            $pluginManager->unloadPlugin($plugin);
+            $pluginManager->uninstallPlugin($plugin);
+
+        } else {
+            $this->makeSurePluginIsRemovedFromFilesystem($plugin);
+        }
+    }
```

Comment:
Do you think it would make sense to have this behavior in the plugin manager? I'm thinking if we need to uninstall a plugin elsewhere we would have to rewrite that code (and maybe it's already the case today?).